x-pack/plugins/security_solution/common/api/detection_engine/prebuilt_rules/model/diff/rule_diff/rule_diff.ts

```diff
@@ -72,6 +76,10 @@ export interface FullRuleDiff {
 export interface PartialRuleDiff {
   fields: Partial<RuleFieldsDiff>;
   has_conflict: boolean;
+  has_non_solvable_conflict: boolean;
+  num_fields_with_updates: number;
+  num_fields_with_conflicts: number;
+  num_fields_with_non_solvable_conflicts: number;
```

Comment:
Nit:

```ts
interface BaseRuleDiff {
  has_conflict: boolean;
  has_non_solvable_conflict: boolean;
  num_fields_with_updates: number;
  num_fields_with_conflicts: number;
  num_fields_with_non_solvable_conflicts: number;
}

/**
 * Full rule diff contains diffs for all the top-level rule fields.
 * Even if there's no change at all to a given field, its diff will be included in this object.
 * This diff can be useful for internal server-side calculations or debugging.
 * Note that this is a pretty large object so returning it from the API might be undesirable.
 */
export interface FullRuleDiff extends BaseRuleDiff {
  fields: RuleFieldsDiff;
}

/**
 * Partial rule diff contains diffs only for those rule fields that have some changes to them.
 * This diff can be useful for returning info from REST API endpoints because its size is tolerable.
 */
export interface PartialRuleDiff extends BaseRuleDiff {
  fields: Partial<RuleFieldsDiff>;
}
```